site/src/docs/patterns/forms/form-validation.mdx

```diff
@@ -239,14 +247,86 @@ The usage of the error summary component is triggered when the form is submitted
 2. The focus is moved from the form submit button to the error summary heading.
 3. The error summary is populated with a list of errors found in the form. Errors are in the same order as they appear in the form. For each error, a running number and an error description are provided. The error description also acts as an anchor link.
 4. The user can use the error description anchor link to quickly jump and focus each of the erroneous inputs.
-5. When the user submits the form again and errors are still found, the contents of the error summary are updated and the focus is moved back to the summary heading again.
-
-<Image
-  src="/images/patterns/form/validation/validation-pattern-example@2x.png"
-  alt="Validation summary example"
-  style={{ maxWidth: 432 }}
-  viewable
-/>
+5. The error validation check happens immediately after the input value changes.
+6. When the user submits the form again and errors are still found, the contents of the error summary are updated and the focus is moved back to the summary heading again.
+
+<Playground scope={{ Button, ErrorSummary, TextInput }} >
+
+```jsx
+import { Button, ErrorSummary, TextInput } from 'hds-react';
+
+() => {
+    const [formValues, setFormValues] = React.useState({});
+    const [formErrors, setFormErrors] = React.useState({});
+    const [isSubmitted, setIsSubmitted] = React.useState(false);
+    const errorList = Object.entries(formErrors);
+
+    const validate = (values, onlyValidationErrors) => {
+      const errors = {
+        ...((!values['firstName'] && !onlyValidationErrors) || (!values['firstName'] && onlyValidationErrors && 'firstName' in formErrors) ? { firstName: 'Please enter your first name' } : {}),
```

Comment:
I guess instead of `'firstName' in formErrors` also` formErrors['firstName']` can be used, because there is no scenario where firstName key exists without an error?

And then I'd simplify the validation with
```

const getFieldError = (fieldName, formData, errorData) => {
return errorData ? !formData[fieldName] && errorData[fieldName] : !formData[fieldName]
}

const validate = (values, currentErrors) => {
const hasFirstNameError = getFieldError('firstName', values, currentErrors)
const hasLastNameError = getFieldError('lastName', values, currentErrors)

const errors = {...(hasFirstNameError ? { firstName: 'Please enter your first name' } : {})}
// and same with hasLastNameError

}
```
And call `validate` with `formErrors` instead of boolean inside `useEffect`.

The error object creation is quite confusing to read at the moment. 😄

Reply:
I haven't made any forms in well over a year and I was Struggling, and it shows 😂